packages/muelu/src/Misc/MueLu_FilteredAFactory_def.hpp

```diff
@@ -501,16 +499,27 @@ namespace MueLu {
 
     // Lists of nodes in each aggregate
     struct {
-      Array<LO> ptr,nodes, unaggregated;
+      // GH: For now, a lot of these routines (e.g. singleton deletion) seem unnecessary to run on device.
```

Comment:
I'd say that they haven't been ported to device, not that that is unnecessary.

Reply:
Well, I think I chose the word `unnecessary` simply because things like singleton deletion seem like low-intensity operations that would likely be less efficient if run on a device.

Reply:
And so it's a good idea to copy everything to host? I disagree.

Reply:
Well, until we set `FilteredAFactory` to run device kernels, we have to copy to host. But I can see how that would change the interpretation of the comment.